### PR TITLE
[dotnet wrapper] Custom wallet storage wrapper

### PR DESCRIPTION
**Pull request by tmarkovski**
This is a long overdue wrapper that was missing since the custom wallet storage was introduced.

- Similar to previous implementation of custom wallet type, this wrapper exposes an interface `IWalletStorage` that implementers can use to hook up a custom storage.
- A minimal in-memory implementation is included with the samples.
- Some cosmetic changes in the samples console reporting
- Package version bumped to 1.6.8

Signed-off-by: Tomislav Markovski <tmarkovski@gmail.com>

### CODE REVIEW

File: wrappers/dotnet/indy-sdk-dotnet/WalletApi/Wallet.cs [hunk 1]
```diff
@@ -13,12 +14,13 @@ namespace Hyperledger.Indy.WalletApi
     /// Represents a wallet that stores key value records and provides static methods for managing
     /// wallets.
     /// </summary>
+    [SuppressMessage("ReSharper", "InconsistentNaming")]
     public sealed class Wallet : IDisposable
```
Comment (mattraffel) [inline wrappers/dotnet/indy-sdk-dotnet/WalletApi/Wallet.cs]: small thing, I feel this code analysis tag for resharper should be removed.  not everyone uses resharper.  unless the intent is to make resharper a requirement for this project.

**Comment by tmarkovski** [comment]
@vimmerru Thanks for the clarification. I can see the reasoning for this.
Is the plan to keep exposing `indy_register_wallet_storage` as C callable method to continue supporting development of non-native plugins in proprietary enterprise scenarios?

**Comment by vimmerru** [comment]
@tmarkovski ```indy_register_wallet_storage``` is public libindy API and will be supported. We just don't want to expose this in wrappers.

**Comment by tmarkovski** [comment]
@vimmerru Great, thank you.

@jovfer Closing this PR. I'll submit another maintenance PR soon with some cleanup code of the previous wallet type implementation.

**Pull request closed without merging**